src/Catalyst.Core.Modules.Dfs/DfsService.cs

```diff
@@ -0,0 +1,663 @@
+            IBootstrapApi bootstrapApi,
+            IConfigApi configApi,
+            IDagApi dagApi,
+            IDhtApi dhtApi,
+            IDnsApi dnsApi,
+            KatDhtService dhtService,
+            IUnixFsApi unixFsApi,
+            IKeyApi keyApi,
+            INameApi nameApi,
+            IObjectApi objectApi,
+            IPinApi pinApi,
+            Ping1 pingService,
+            IPubSubApi pubSubApi,
+            PubSubService pubSubService,
+            IStatsApi statsApi,
+            ISwarmApi swarmApi,
+            SwarmService swarmService,
+            DfsOptions dfsOptions,
+            IHashProvider hashProvider,
+            DfsState dfsState,
+            IPasswordManager passwordManager)
+        {
+            BitSwapApi = bitSwapApi;
+            BitSwapService = bitSwapService;
+            BlockApi = blockApi;
+            BlockRepositoryApi = blockRepositoryApi;
+            BootstrapApi = bootstrapApi;
+            ConfigApi = configApi;
+            DagApi = dagApi;
+            DhtApi = dhtApi;
+            DhtService = dhtService;
+            UnixFsApi = unixFsApi;
+            KeyApi = keyApi;
+            NameApi = nameApi;
+            ObjectApi = objectApi;
+            PinApi = pinApi;
+            PingService = pingService;
+            PubSubApi = pubSubApi;
+            PubSubService = pubSubService;
+            StatsApi = statsApi;
+            SwarmApi = swarmApi;
+            SwarmService = swarmService;
+            Options = dfsOptions;
+            _hashProvider = hashProvider;
+            _dfsState = dfsState;
+            DnsApi = dnsApi;
+
+            _passphrase = passwordManager.RetrieveOrPromptAndAddPasswordToRegistry(PasswordRegistryTypes.IpfsPassword,
+                "Please provide your IPFS password");
+
+            InitAsync().Wait();
+        }
+
+        internal virtual AddFileOptions AddFileOptions()
+        {
+            return new AddFileOptions
+            {
+                Hash = _hashProvider.HashingAlgorithm.Name,
+                RawLeaves = true
+            };
+        }
+
+        /// <summary>
+        ///     The configuration options.
+        /// </summary>
+        public DfsOptions Options { get; set; }
+
+        private IKeyStoreService _keyStoreService;
+
+        private async Task InitAsync()
+        {
+            MigrationManager = new MigrationManager(Options.Repository);
+
+            Log.Debug("Building local peer");
+            Log.Debug("Getting key info about self");
+            await KeyApi.SetPassphraseAsync(_passphrase).ConfigureAwait(false);
+
+            var self = await KeyApi.GetPublicKeyAsync("self").ConfigureAwait(false)
+             ?? await KeyApi.CreateAsync("self", "rsa", 0).ConfigureAwait(false);
+
+            var localPeer = new Peer
+            {
+                Id = self.Id,
+                PublicKey = await (await KeyChainAsync()).GetPublicKeyAsync("self").ConfigureAwait(false),
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add '.ConfigureAwait(false)' to this call to allow execution to continue in any thread.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)